### ISSUE TITLE
[Bug]: WAS - Wolfsburger Abfallwirtschaft u. Straßenreinigung did no longer collects the waste appointments

### ISSUE THREAD

**Issue by dp20eic** [mampfes/hacs_waste_collection_schedule#1778]
### I Have A Problem With:

A specific source

### What's Your Problem

Since February 8th, 2024, I no longer see any appointments displayed in the calendar and therefore no more appointments are shown to me in the automations.

## my configuration
```
# Abfallkallender
waste_collection_schedule:
  sources:
    - name: was_wolfsburg_de
      args:
        city: my district
        street: my street
  fetch_time: "04:00"
  day_switch_time: "10:00"
sensor:
  # ------- Garbage Collection -------
  # next collection
  - platform: waste_collection_schedule
    name: "Nächste Abholung"
    value_template: '{{value.types|join(", ")}}{% if value.daysTo == 0 %} Heute{% elif value.daysTo == 1 %} Morgen{% else %} in {{valu
  # Nächste Bioabfall Abholung
  - platform: waste_collection_schedule
    name: "Bioabfall"
    value_template: '{{value.types|join(", ")}}{% if value.daysTo == 0 %} Heute{% elif value.daysTo == 1 %} Morgen{% else %} in {{valu
    types:
      - Bioabfall
  # Nächste gelber-Sack Abholung
  - platform: waste_collection_schedule
    name: "Gelber Sack"
    value_template: '{{value.types|join(", ")}}{% if value.daysTo == 0 %} Heute{% elif value.daysTo == 1 %} Morgen{% else %} in {{valu
    types:
      - Gelber Sack
  # Nächste Papierabfall Abholung
  - platform: waste_collection_schedule
    name: "Altpapier"
    value_template: '{{value.types|join(", ")}}{% if value.daysTo == 0 %} Heute{% elif value.daysTo == 1 %} Morgen{% else %} in {{valu
    types:
      - Altpapier
  # Nächste Restabfall-Abholung
  - platform: waste_collection_schedule
    name: "Restabfall "
    value_template: '{{value.types|join(", ")}}{% if value.daysTo == 0 %} Heute{% elif value.daysTo == 1 %} Morgen{% else %} in {{valu
    types:
      - Restabfall
```

### Relevant Configuration

```YAML
I saw that WAS changed the website, that could be the reason that the data can no longer be found.
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by ppk2000** [comment]
Hello, 

Yes, the WAS website has changed. There are the news URL:

Gelber Sack:
r = requests.post(
            "https://was-wolfsburg.de/wp-content/themes/astra-child/search_gelbersack/search.php",
            data={"loc": "MY_CITY"},
            headers={"content-type": "application/x-www-form-urlencoded; charset=UTF-8"},
        )
		

Bioabfall,Restabfall,Altpapier:
r = requests.post(
            "https://was-wolfsburg.de/wp-content/themes/astra-child/search_restabfall/search.php",
            data={"loc": "MY_STREET"},
            headers={"content-type": "application/x-www-form-urlencoded; charset=UTF-8"},

        )

**Comment by ppk2000** [comment]
ICS don´t run in WAS website anymore.
You can do a list like the Gelber Sack.

I´ve done :

 r = requests.post(
            "https://was-wolfsburg.de/wp-content/themes/astra-child/search_restabfall/search.php",
            data={"loc": "YOUR_STREET"},
            headers={"content-type": "application/x-www-form-urlencoded; charset=UTF-8"},
        )

        entries = []
        match = re.findall(r"(\d{2})\.(\d{2})\.(\d{4})", r.text)
        match2 = re.findall(r"Bioabfall|Restabfall|Altpapier", r.text)
        myit = iter(match2)
        for m in match:
            date = datetime.date(day=int(m[0]), month=int(m[1]), year=int(m[2]))
            entries.append(Collection(date, next(myit)))

VG

**Comment by Dark-Tower-Coder** [comment]
I would also be very happy about an adjustment.

**Comment by boehser-enkel** [comment]
Me too

**Comment by boehser-enkel** [comment]
> > Me too (also where is the ics for the dates?)
> 
> Good morning, I don't understand that, it has nothing to do with the integration, the integration queries the data directly on the WAS page. If you want to save ICS data into a calendar and read it out, then WAS has to have an ICS calendar provide, or you enter the appointments yourself into a calendar, and then you have to make your adjustments in the `.yaml`, which comes from the calendar you created. As I know, there is no ICS file anymore you can download from the WAS page.
> 
> VG Bernd

The ics is coming

**Comment by boehser-enkel** [comment]
> Moin,
> 
> Today I was offered an update of the `hacs_waste_collection_schedule`, unfortunately I found that the error was not fixed :) Ok, I change it again at original Source and wait if it changed next update.
> 
> VG

Maybe i have an error in there. Could you upload your complete was_wolfsburg_de.py please?

**Comment by FelixArndt** [comment]
Yeah, would be awesome if someone can fix this :-)

**Comment by ppk2000** [comment]
The URLs works fine for me. Maybe the problem are last changes in the integration???
Sorry but I use my own implementation  and i can´t help.

**Comment by slashphotos** [comment]
Hey!

Is there any update to this issue?
I can't get it to work. Neither the master nor this code snippet are working in homeassistant or in my python environment. If I try to access the URL by hand, I get 
`{"status":true,"count":0,"string":null}`
as a response.

Kind regards

**Comment by 5ila5** [comment]
fixed in #2292

This will be part of the next release or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
